packages/blockchain-link-types/src/solana.ts

```diff
@@ -1,24 +1,73 @@
-import type { ParsedTransactionWithMeta } from '@solana/web3.js';
+import type {
+    AccountInfoBase,
+    AccountInfoWithBase58EncodedData,
+    AccountInfoWithBase64EncodedData,
+    AccountInfoWithBase64EncodedZStdCompressedData,
+    AccountInfoWithJsonData,
+    GetTransactionApi,
+    Signature,
+} from '@solana/web3.js';
+
+import type {
+    GetObjectWithKey,
+    GetObjectWithoutKey,
+    ObjectsOnly,
+    Overloads,
+} from '@trezor/type-utils';
 
-export type SolanaValidParsedTxWithMeta = ParsedTransactionWithMeta & {
-    meta: Required<NonNullable<ParsedTransactionWithMeta['meta']>>;
-    transaction: Required<ParsedTransactionWithMeta['transaction']>;
-    blockTime: Required<NonNullable<ParsedTransactionWithMeta['blockTime']>>;
+type GetTransactionApiOverloads = Overloads<GetTransactionApi['getTransaction']>;
+type GetJsonParsedTransactionApiOverloads = {
+    [P in keyof GetTransactionApiOverloads]: GetTransactionApiOverloads[P] extends (
+        ...args: [
+            signature: Signature,
+            config: Readonly<{
+                encoding: 'jsonParsed';
```

Comment:
This is intense, but basically boils down to: choose the overload that has `encoding: 'jsonParsed'`, infer its return type, and use that.